torch/_functorch/aot_autograd.py

```diff
@@ -3247,7 +3247,20 @@ def runtime_wrapper(*args):
                     if trace_joint:
                         assert isinstance(updated_inpt, TensorAlias)
                         updated_inpt = updated_inpt.alias
-                    original_inpt.set_(updated_inpt)
+                    try:
+                        original_inpt.set_(updated_inpt)
+                    except RuntimeError:
+                        # This happens when the sizes of the two tensors do not match, probably
+                        # because we saw .data
+                        # while we could call
+                        # original_inpt.data = updated_inpt
+                        # That's a little gross, let's use the "safe" pattern instead
+                        with torch.no_grad():
```

Comment:
What happens if we unconditionally run this `set_` under `no_grad`?

Reply:
Not sure, let me check.

Reply:
It fails